internal/cluster/distribution/eks/eksprovider/workflow/select_volume_size_activity.go

```diff
@@ -0,0 +1,70 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workflow
+
+import (
+	"context"
+	"fmt"
+	"math"
+
+	"emperror.dev/errors"
+)
+
+const (
+	SelectVolumeSizeActivityName = "eks-select-volume-size-activity"
+)
+
+type SelectVolumeSizeActivity struct {
+	defaultVolumeSize int
+	unsetVolumeSize   int
+}
+
+type SelectVolumeSizeActivityInput struct {
+	EKSActivityInput
+	AMIImageSize int
+}
+
+type SelectVolumeSizeActivityOutput struct {
+	VolumeSize int
+}
+
+func NewSelectVolumeSizeActivity(defaultVolumeSize, unsetVolumeSize int) (activity *SelectVolumeSizeActivity) {
+	return &SelectVolumeSizeActivity{
+		defaultVolumeSize: defaultVolumeSize,
+		unsetVolumeSize:   unsetVolumeSize,
+	}
+}
+
+func (activity *SelectVolumeSizeActivity) Execute(ctx context.Context, input SelectVolumeSizeActivityInput) (output *SelectVolumeSizeActivityOutput, err error) {
+	if input.AMIImageSize == 0 {
+		return nil, errors.New("invalid zero AMI image size")
+	}
+
+	output = &SelectVolumeSizeActivityOutput{}
+	if activity.defaultVolumeSize != 0 {
+		output.VolumeSize = activity.defaultVolumeSize
+	} else {
+		output.VolumeSize = int(math.Max(float64(activity.unsetVolumeSize), float64(input.AMIImageSize)))
```

Comment:
I'd use a local max(int,int) function or an if statement. I have bad feelings about floating point arithmetics in such context. (Up to you.)

Reply:
What do you mean by this, what's the problem with the floating point arithmetic?

I know it cannot be used for equality comparison properly, but otherwise the ordering of values is supposed to be deterministic AFAIK so what I see here as worst case is that it picks an arbitrary value when the two values are actually equal (which sounds okay to me).

(I always use standard library over custom definition when it's possible and doesn't impose too much extra work, these three constructions are stretching the boundary, but it might make the reader's job easier who would otherwise have to inspect the custom function as well so I'm curios whether there is any actual logical problem here which would warrant me not to ever use this approach again.)